streams/src/test/java/org/apache/kafka/streams/integration/KStreamRepartitionIntegrationTest.java

```diff
@@ -90,6 +95,22 @@
     private Properties streamsConfiguration;
     private List<KafkaStreams> kafkaStreamsInstances;
 
+    @Parameters(name = "Optimization = {0}")
+    public static Collection<Object[]> data() {
+        final List<Object[]> values = new ArrayList<>();
+
+        Arrays.asList(StreamsConfig.OPTIMIZE, StreamsConfig.NO_OPTIMIZATION)
+              .forEach(x -> values.add(new Object[]{x}));
+
+        return values;
+    }
+
+    public KStreamRepartitionIntegrationTest(final String topologyOptimization) {
```

Comment:
A simple
```
    @Parameter
    public String topologyOptimization;
```

Would be sufficient instead of adding a constructor and those lines could go into `before()`.

(As above, feel free to ignore this comment.)